src/main/java/com/internship/backend/service/UserService.java

```diff
@@ -4,119 +4,123 @@
-        if (userRepository.findByUsername(user.getUsername()) != null) {
+        if (userRepository.findByUsername(user.getUsername()).isPresent()) {
             throw new UserAlreadyExistsException("Username already exists");
         }
-        if (userRepository.findByEmail(user.getEmail()) != null) {
+        if (userRepository.findByEmail(user.getEmail()).isPresent()) {
             throw new UserAlreadyExistsException("Email already exists");
         }
+
+        user.setId(idGeneratorService.getCurrentId());
         String hashPassword = passwordEncoder.encode(user.getPassword());
         user.setPassword(hashPassword);
 
         Users savedUser = userRepository.save(user);
 
         Set<Authority> authorities = user.getAuthorities();
         Iterator<Authority> iterator = authorities.iterator();
-
         authorityRepository.save(iterator.next());
 
         return savedUser;
     }
 
     public Users fromDTO(UserDTO userDTO) {
-        Users user = Users.builder()
-                .username(userDTO.getUsername())
-                .password(userDTO.getPassword())
-                .email(userDTO.getEmail())
-                .build();
-
-        Set<Authority> authorities = new HashSet<>();
-
-        Authority authority = new Authority();
-        authority.setName(userDTO.getRole());
-        authority.setUser(user);
+        Users user = new Users();
+        user.setUsername(userDTO.getUsername());
+        user.setPassword(userDTO.getPassword());
+        user.setEmail(userDTO.getEmail());
 
-        authorities.add(authority);
-        user.setAuthorities(authorities);
+        Authority authority = resolveAuthority(userDTO.getRole());
+        user.setAuthorities(Collections.singleton(authority));
 
         return user;
     }
 
-    public Users update(int userId, Users updatedUser) throws UserDoesNotExistException {
-        Users user = userRepository.findById(userId).orElseThrow(() -> new UserDoesNotExistException("User not found"));
+    private Authority resolveAuthority(String roleName) {
+        Optional<Authority> optionalAuthority = authorityRepository.findByName(roleName);
+        if (optionalAuthority.isPresent()) {
+            return optionalAuthority.get();
+        } else {
+            Authority newAuthority = new Authority(roleName);
+            newAuthority.setId(idGeneratorService.getCurrentId());
+            return authorityRepository.save(newAuthority);
+        }
+    }
 
-        user.setUsername(updatedUser.getUsername());
-        user.setPassword(updatedUser.getPassword());
-        user.setEmail(updatedUser.getEmail());
 
-        return userRepository.save(user);
+    public List<Users> getAllUsers() {
+        List<Users> users = userRepository.findAll();
+
+        users.forEach(user -> {
+            List<Reservation> reservations = reservationRepository.findByUserId(user.getId());
+            user.setReservations(reservations);
```

Comment:
I think that this endpoint only aims to find all users, there is no need to search for all their reservations